drivers/sensor/si7006/si7006.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2019 Electronut Labs
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sensor.h>
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <string.h>
+#include <sys/byteorder.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+#include <i2c.h>
+#include <logging/log.h>
+#include <stdio.h>
+#include "si7006.h"
+
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+LOG_MODULE_REGISTER(si7006);
+
+struct si7006_data {
+	struct device *i2c_dev;
+	u16_t temperature;
+	u16_t humidity;
+};
+
+/**
+ * @brief function to get relative humidity
+ *
+ * @return int 0 on success
+ */
+static int si7006_get_humidity(struct device *i2c_dev,
+	struct si7006_data *si_data)
+{
+	int retval = 0;
+	uint8_t hum[2];
+
+	if (i2c_burst_read(i2c_dev, DT_INST_0_SILABS_SI7006_BASE_ADDRESS,
+			SI7006_MEAS_REL_HUMIDITY_MASTER_MODE, hum, sizeof(hum))
+			!= 0) {
+		LOG_ERR("read register err");
+		retval = -1;
+	}
+
+	si_data->humidity = (hum[0] << 8) | (hum[1]);
```

Comment:
I like strong parenthesization to avoid ambiguity, but around `hum[1]` it doesn't help.  Nobody's going to confuse this with `((hum[0] << 8) | hum)[1]`.

Also in temperature function.